Core/Core/Logging/SpeckleLog.cs

```diff
@@ -0,0 +1,210 @@
+    /// </summary>
+    /// <param name="hostApplicationName">Name of the application using this SDK ie.: "Rhino"</param>
+    /// <param name="hostApplicationVersion">Public version slug of the application using this SDK ie.: "2023"</param>
+    /// <param name="logConfiguration">Input configuration object.</param>
+    /// <returns>Logger instance</returns>
+    public static Serilog.Core.Logger CreateConfiguredLogger(
+      string hostApplicationName,
+      string? hostApplicationVersion,
+      SpeckleLogConfiguration logConfiguration
+    )
+    {
+      // TODO: check if we have write permissions to the file.
+      // if not, disable file sink, even if its enabled in the config
+      // show a warning about that...
+      var canLogToFile = true;
+      var logFilePath = Path.Combine(
+        SpecklePathProvider.LogFolderPath(hostApplicationName, hostApplicationVersion),
+        "SpeckleCoreLog.txt"
+      );
+      var serilogLogConfiguration = new LoggerConfiguration().MinimumLevel
+        .Is(logConfiguration.minimumLevel)
+        .Enrich.FromLogContext()
+        .Enrich.FromGlobalLogContext()
+        .Enrich.WithExceptionDetails();
+
+      if (logConfiguration.logToFile && canLogToFile)
+        serilogLogConfiguration = serilogLogConfiguration.WriteTo.File(
+          logFilePath,
+          rollingInterval: RollingInterval.Day,
+          retainedFileCountLimit: 10
+        );
+
+      if (logConfiguration.logToConsole)
+        serilogLogConfiguration = serilogLogConfiguration.WriteTo.Console();
+
+      if (logConfiguration.logToSeq)
+        serilogLogConfiguration = serilogLogConfiguration.WriteTo.Seq(
+          "https://seq.speckle.systems",
+          apiKey: "1bF86pB8XI3s4pYtc2kp"
+        );
+
+      if (logConfiguration.logToSentry)
+        serilogLogConfiguration = serilogLogConfiguration.WriteTo.Sentry(o =>
+        {
+          o.Dsn =
+            "https://94275909c1094f2388224c9222b0cfba@o436188.ingest.sentry.io/4504491155783680";
+          o.Debug = true;
+          // Set traces_sample_rate to 1.0 to capture 100% of transactions for performance monitoring.
+          // We recommend adjusting this value in production.
+          o.TracesSampleRate = 1.0;
+          // Enable Global Mode if running in a client app
+          o.IsGlobalModeEnabled = true;
+          // Debug and higher are stored as breadcrumbs (default is Information)
+          o.MinimumBreadcrumbLevel = LogEventLevel.Debug;
+          // Warning and higher is sent as event (default is Error)
+          o.MinimumEventLevel = LogEventLevel.Warning;
+        });
+
+      var logger = serilogLogConfiguration.CreateLogger();
+      if (logConfiguration.logToFile && !canLogToFile)
+        logger.Warning("Log to file is enabled, but cannot write to {LogFilePath}", logFilePath);
+      return logger;
+    }
+
+    private static void _addUserIdToGlobalContextFromDefaultAccount()
+    {
+      var id = "unknown";
```

Comment:
we could maybe try to take another unique identifier as a mac address or machine name, to help us understand how many users might be affected by a specific issue